res/res_rtp_asterisk.c

```diff
@@ -9191,6 +9208,14 @@ static int rtp_red_buffer(struct ast_rtp_instance *instance, struct ast_frame *f
 
 	if (frame->datalen > 0) {
 		if (red->t140.datalen > 0) {
+			/*
+			* Avoid merging command and regular T.140 text packets by flushing the
+			* previous T.140 packets. When realtime text packets are to be sent, the
+			* text is accumulated in a buffer and sent regularly by a timer.  It can
+			* happen that commands such as a backspace, CR, or LF get merged with
+			* regular text. This breaks some UAs. In recent tests it could not be
+			* reproduced, but its probably better to keep it.
+			*/
```

Comment:
After reading the RFC, specifically [Section 3.3](https://www.rfc-editor.org/rfc/rfc4103#section-3.3):

> The T140block contains one or more T.140 code elements as specified in [Recommendation T.140].

I am of the opinion that this change was incorrect when it went in. If a UA can't handle more than t code element, it is broken, not Asterisk.

Not asking you to change anything, just venting.

Reply:
Yeah, classic case unfortunately.